pkgs/by-name/ke/key/package.nix

```diff
@@ -19,20 +22,35 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "key";
-  version = "2.12.2";
+  version = "2.12.3";
   src = fetchFromGitHub {
     owner = "KeYProject";
     repo = "key";
-    tag = "KeY-${version}";
-    hash = "sha256-veqaWyWEiTot2cAjvyPG+Ra8/pqS4i6w6iR+qhozIM4=";
+    tag = "KEY-${version}";
+    hash = "sha256-1pN0lmr/teVitpMIM9M9lSTkmnVcZwdAQay2pzgJDCk=";
   };
 
+  patches = [
+    # Remove linting framework, causes issues with the update script.
+    ./remove-eisop-checker.patch
+  ];
+
+  # Continuation of the patch above.
+  postPatch = ''
+    substituteInPlace build.gradle \
+      --replace-fail 'id "org.checkerframework"' '//' \
+      --replace-fail 'apply plugin: "org.checkerframework"' "" \
+      --replace-fail 'checkerFramework' "//" \
+      --replace-fail 'def anchor = "git merge-base HEAD origin/main".execute().getText()' 'return []'
+    substituteInPlace key.core/build.gradle \
+      --replace-fail '"git rev-parse $args".execute().text.trim()' '"${version}"'
```

Comment:
Why not a patch?

Reply:
The last replace uses `${version}`. I suppose I could put the other stuff in the patch.

Reply:
I think it'd be clearer in a patch, yes.

You could even include the last one if you do something like this:

https://github.com/NixOS/nixpkgs/blob/6421090605340b29544ff9fc9e6f5a850fe759f8/pkgs/games/shattered-pixel-dungeon/summoning-pixel-dungeon/default.nix#L21-L28

https://github.com/NixOS/nixpkgs/blob/6421090605340b29544ff9fc9e6f5a850fe759f8/pkgs/games/shattered-pixel-dungeon/summoning-pixel-dungeon/disable-git-version.patch#L26